kolibri/core/assets/src/conditionalPromise.js

```diff
@@ -0,0 +1,51 @@
+class ConditionalPromise {
+  /**
+   * Create a conditional promise - like a promise, but cancelable!
```

Comment:
I like the approach you took, but it might be misleading to say here that the promise itself is "cancelable", since there's no way to cancel it such that normal `.then` callbacks won't fire. Perhaps "like a promise, but with the option of conditional callbacks" or something? Not as zingy...